modules/ballerina-http/src/main/ballerina/ballerina/net/http/resiliency/circuit-breaker.bal

```diff
@@ -0,0 +1,368 @@
+                http:Response errorResponse = createErrorResponse(e.msg, e.statusCode);
+                return errorResponse, e;
+            }
+        } else {
+            // TODO: Allow the user to handle this scenario. Maybe through a user provided function
+            http:Response errorResponse = handleOpenCircuit(circuitHealth, resetTimeout);
+            return errorResponse, e;
+        }
+
+        return res, null;
+    }
+
+    action get (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        currentState = updateCircuitState(circuitHealth, currentState, failurePercentageThreshold, resetTimeout);
+        http:Response res;
+        http:HttpConnectorError e;
+
+        if (currentState == CircuitState.CLOSE) {
+            res, e = httpEP.get(path, req);
+            circuitHealth.requestCount = circuitHealth.requestCount + 1;
+
+            if (e != null) {
+                circuitHealth.updateHealth();
+                http:Response errorResponse = createErrorResponse(e.msg, e.statusCode);
+                return errorResponse, e;
+            }
+        } else if (currentState == CircuitState.HALF_OPEN) {
+            res, e = httpEP.get(path, req);
+            circuitHealth.requestCount = circuitHealth.requestCount + 1;
+
+            if (e != null) {
+                circuitHealth.updateHealth();
+                http:Response errorResponse = createErrorResponse(e.msg, e.statusCode);
+                return errorResponse, e;
+            }
+        } else {
+            // TODO: Allow the user to handle this scenario. Maybe through a user provided function
+            http:Response errorResponse = handleOpenCircuit(circuitHealth, resetTimeout);
+            return errorResponse, e;
+        }
+
+        return res, null;
+    }
+}
+
+function updateCircuitState (CircuitHealth circuitHealth, CircuitState currentState, float failureThreshold,
+                             int resetTimeout) (CircuitState) {
+    if (currentState == CircuitState.OPEN) {
+        Time currentT = currentTime();
+        int elapsedTime = currentT.time - circuitHealth.lastErrorTime.time;
+
+        if (elapsedTime > resetTimeout) {
+            circuitHealth.errorCount = 0;
+            circuitHealth.requestCount = 0;
+            currentState = CircuitState.HALF_OPEN;
+        }
+    } else if (currentState == CircuitState.HALF_OPEN) {
+        if (circuitHealth.errorCount > 0) {
+            // If the trial run has failed, trip the circuit again
+            currentState = CircuitState.OPEN;
+        } else {
+            // If the trial run was successful reset the circuit
+            currentState = CircuitState.CLOSE;
+        }
+    } else {
+        float currentFailureRate;
+
+        if (circuitHealth.requestCount > 0) {
+            currentFailureRate = (circuitHealth.errorCount / circuitHealth.requestCount) * 100;
+        } else {
+            currentFailureRate = 0;
+        }
```

Comment:
Shall we remove this else block by initializing currentFailureRate to 0, which is more cleaner?